flexget/plugins/sites/newznab.py

```diff
@@ -1,125 +1,615 @@
+    'xxx/web-dl': 6090,  # unofficial
+    # Misc
+    'misc': 7000,
+    'misc/misc': 7010,
+    'misc/ebook': 7020,
+    'misc/comics': 7030,
+    # Unknown
+    'not-determined': 7900
+}
+
+field_map = {
+    'title': 'title',
+    'url': 'link',
+    'newznab_pubdate': lambda xml: value_to_naive_utc(find_value(['updated_parsed', 'updated'], xml)),
+    'newznab_grabs': lambda xml: value_to_int(find_value('newznab_grabs.value', xml)),
+    'newznab_size': lambda xml: value_to_int(find_value('newznab_size.value', xml)),
+    'newznab_files': lambda xml: value_to_int(find_value('newznab_files.value', xml)),
+    'newznab_usenet_date': lambda xml: value_to_naive_utc(find_value('newznab_usenetdate.value', xml)),
+    'newznab_password': lambda xml: value_to_int(find_value('newznab_password.value', xml)),
+    'newznab_guid': lambda xml: re.split(r'[/=]', find_value(['id', 'guid'], xml, default='')).pop(),
+    'newznab_usenet_group': 'newznab_group.value',
+    'newznab_tvairdate': lambda xml: value_to_naive_utc(find_value('newznab_tvairdate.value', xml)),
+    'newznab_season':
+        lambda xml: value_to_int(find_value('newznab_season.value', xml, regex=r'[Ss]([0-9]{1,2})')),
+    'newznab_episode':
+        lambda xml: value_to_int(find_value('newznab_episode.value', xml, regex=r'[Ee]([0-9]{1,3})')),
+    'newznab_hydra_indexer_name': 'newznab_hydraindexername.value',
+    'newznab_hydra_indexer_host': 'newznab_hydraindexerhost.value',
+    'newznab_hydra_indexer_score': lambda xml: value_to_int(find_value('newznab_hydraindexerscore.value', xml))
+}
+
+field_validation_list = [
+    'title',
+    'url'
+]
+
+PLUGIN_LOOKUP_MAP = {
+    'tvdb': 'thetvdb_lookup',
+    'trakt': 'trakt_lookup',
+    'tvmaze': 'tvmaze_lookup',
+    'imdb': 'imdb_lookup',
+    'tmdb': 'tmdb_lookup'
+}
+
+NAMESPACE_NAME = 'newznab'
+NAMESPACE_URL = 'http://www.newznab.com/DTD/2010/feeds/attributes/'
+NAMESPACE_TAGNAME = 'attr'
+ENCLOSURE_TYPE = 'application/x-nzb'
+
+
+# list utils
+def list_insert_unique(in_list, idx, element, caseinsensitive=False):
+    if not isinstance(in_list, list) or element is None:
+        return
+    if isinstance(element, str):
+        if element.isspace() or element in in_list:
+            return
+        elif caseinsensitive is True and element.lower() in in_list:
+            return
+    elif element in in_list:
+        return
+    in_list.insert(idx, element)
+
+
+def list_append_unique(in_list, element, caseinsensitive=False):
+    list_insert_unique(in_list, len(in_list), element, caseinsensitive=caseinsensitive)
+
+
+def list_combine_unique(list_dst, list_source, caseinsensitive=False):
+    if not isinstance(list_dst, list) or not isinstance(list_source, list):
+        log.error('Cant combine lists, src/dst is not a listtype: %s, %s', list_dst, list_source)
+    for item in list_source:
+        list_append_unique(list_dst, item, caseinsensitive=caseinsensitive)
+
+
+def _debug_dump_entry(entry):
```

Comment:
What are you talking about? I want to output formatted debug information, this has nothing to-do with breakpoints.